invenio_users_resources/services/generators.py

```diff
@@ -108,3 +109,57 @@ def query_filter(self, identity=None, **kwargs):
                     return dsl.Q("term", id=need.value)
 
         return []
+
+
+class IfNotManaged(Generator):
```

Comment:
Maybe I am asking too much, but just in case you have some time :)
This is already implemented [3 times](https://github.com/search?q=org%3Ainveniosoftware+%22g.needs%28record%3Drecord%2C+**kwargs%29+for+g+in+self._generators%28record%22&type=code) at least, and there is actually an issue for that, see [here](https://github.com/inveniosoftware/invenio-rdm-records/issues/1226).

It would be nice:
1. address https://github.com/inveniosoftware/invenio-rdm-records/issues/1226, by moving the generic ones in `invenio-records-permissions`, where all the base generators are, making sure to add the raise `NotImplementedError` to the not implemented method
2. use it here and in other places.